lib/operations/connect.js

```diff
@@ -212,6 +212,12 @@ function connect(mongoClient, url, options, callback) {
       delete _finalOptions.db_options.auth;
     }
 
+    // `journal` should be translated to `j` for the driver
+    if (_finalOptions.journal != null) {
+      _finalOptions.j = _finalOptions.journal;
```

Comment:
to me it seems like `j` is the odd man out here, shouldn't we filter that out and depend on `journal` throughout the codebase? `j` is just the field on the command send to the server. 

**UPDATE:** I now see that this is a pretty small change, while what I'm proposing would potentially touch much more of the codebase. Maybe we can defer my suggestion here to @reggi s work on `MongoClientOptions`

Reply:
Yeah I think this is the minimal change necessary to fix this issue; it's probably not worth diving deep on this now when we're already planning on overhauling this entire system; definitely worth addressing when we do the overhaul.

Reply:
btw I think the original complaint is for 3.5, so we'll need a port there too

Reply:
#2359